dist/src/main/java/io/camunda/zeebe/gateway/GatewayModuleConfiguration.java

```diff
@@ -110,6 +121,53 @@ public Gateway gateway() {
     return gateway;
   }
 
+  @Bean
+  @ConditionalOnMissingBean(value = RestGatewayDisabled.class)
+  public ResponseObserverProvider responseObserverProvider() {
+    return JobActivationRequestResponseObserver::new;
+  }
+
+  @Bean
+  @ConditionalOnMissingBean(value = RestGatewayDisabled.class)
```

Comment:
:question: Shouldn't this be the opposite condition?

If a `RestGatewayDisabled` bean is missing, it means that an embedded Gateway is enabled in the Broker. In that case, the Gateway doesn't need to create an activate jobs handler since one will be created by the `BrokerModuleConfiguration`.

Reply:
If I'm not mistaken, it means the following: if the `RestGatewayDisabled` bean is missing, the REST gateway is enabled. In that case, we also need those beans here, so the REST controllers (which are enabled in the same way) can inject the dependencies they need.

Reply:
Does that mean, as a result, that the condition in the `BrokerModuleConfigration` is wrong then? So it should be `@ConditionalOnBean(value = RestGatewayDisabled.class)` instead of `@ConditionalOnMissingBean(value = RestGatewayDisabled.class)` there? 🤔

Reply:
@koevskinikola and I clarified this offline:
- Under the `"gateway"` profile (triggering `GatewayModuleConfiguration`), the REST API is always enabled. The related beans can be created without condition.
- Under the `"broker"` profile (triggering `BrokerModuleConfigration`), the REST API is enabled by default (because the [EmbeddedGatewayCfg](https://github.com/camunda/camunda/blob/ff8dbe135d1523283ba1324ed42c98824150432d/zeebe/broker/src/main/java/io/camunda/zeebe/broker/system/configuration/EmbeddedGatewayCfg.java#L16) is enabled by default). It can be disabled using Spring properties, for example. Thus, the related REST API beans should only be created when `@ConditionalOnRestGatewayEnabled`.